arrow-core-data/src/main/kotlin/arrow/core/Ordering.kt

```diff
@@ -3,13 +3,13 @@ package arrow.core
 import arrow.typeclasses.Eq
 import arrow.typeclasses.Hash
 import arrow.typeclasses.Monoid
-import arrow.typeclasses.Order
 import arrow.typeclasses.Semigroup
 import arrow.typeclasses.ShowDeprecation
 import arrow.typeclasses.hashWithSalt
 
-sealed class Ordering {
-  override fun equals(other: Any?): Boolean = this === other // ref equality is fine because objects should be singletons
+sealed class Ordering(private val toInt: Int) {
```

Comment:
👌

Reply:
I'm wondering if we still need this type, or we should deprecate and remove it alongside `Order`.
You could use it in combination with `compareTo` to get a typed API instead of an `Int` based one.

```kotlin
val ordering = 2.right().compareTo(3.right())
  .let(Ordering.Companion::fromInt)
  
when(ordering) {
  is LT -> ...
  is GT -> ...
  is EQ -> ...
}
```

cc\\ @raulraja

Reply:
I think it's a useful utility if you don't want to deal with integers and want a typed api but in practice the purpose of the `compareTo` operator I interpret as a way for users to already syntactically enjoy `<`, `>`, and `==` which already covers all Boolean path cases those types cover. After all those types don't hold any data either. I personally think Ordering has no value but also I think it's harmless o happy to keep it as utility if anyone cares enough.